compiler/src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -50,6 +50,18 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
 
   val containerFlagsMask = Flags.Method | Flags.Lazy | Flags.Accessor | Flags.Module
 
+  /** A map of lazy values to the fields they should null after initialization. */
+  private[this] var lazyValNullables: Map[Symbol, List[Symbol]] = _
+  private def nullableFor(sym: Symbol)(implicit ctx: Context) =
+    if (sym.is(Flags.Module)) Nil
+    else lazyValNullables.getOrElse(sym, Nil)
+
+
+  override def prepareForUnit(tree: Tree)(implicit ctx: Context) = {
+    lazyValNullables = ctx.collectNullableFieldsPhase.asInstanceOf[CollectNullableFields].lazyValNullables
```

Comment:
Ideally we should `clear()` this map once we're done with it, this can be done by overriding `transformUnit`.